nixos/modules/services/web-servers/phpfpm/default.nix

```diff
@@ -296,7 +297,7 @@ in
             ProtectHome = true;
             # XXX: We need AF_NETLINK to make the sendmail SUID binary from postfix work
             RestrictAddressFamilies = "AF_UNIX AF_INET AF_INET6 AF_NETLINK";
-            Type = "notify";
+            Type = if cfg.settings.systemd_interval != 0 then "notify-reload" else "notify";
             ExecStart = "${poolOpts.phpPackage}/bin/php-fpm -y ${cfgFile} -c ${iniFile}";
             ExecReload = "${pkgs.coreutils}/bin/kill -USR2 $MAINPID";
```

Comment:
iiuc this is covered by the built-in behavior
> Behavior of notify-reload is similar to notify, with one difference: the SIGHUP UNIX process signal is sent to the service's main process when the service is asked to reload and the manager will wait for a notification about the reload being finished.

https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#Type=

```suggestion
```

Reply:
the documentation you quoted specifies `SIGHUP` but `php-fpm` requires `SIGUSR2`, hence the custom `ExecReload`

Reply:
ho ho i hadn't seen that one yet! thanks @Ma27 :heart: